Mathlib/AlgebraicTopology/SimplexCategory/Defs.lean

```diff
@@ -167,6 +176,19 @@ noncomputable def inclusion.fullyFaithful (n : ℕ) :
 theorem Hom.ext {n} {a b : Truncated n} (f g : a ⟶ b) :
     f.toOrderHom = g.toOrderHom → f = g := SimplexCategory.Hom.ext _ _
 
+open Mathlib.Tactic (subscriptTerm) in
+/-- For `m ≤ n`, `⦋m⦌ₙ` is the `m`-dimensional simplex in `Truncated n`. The
+proof `p : m ≤ n` can also be provided using the syntax `⦋m, p⦌ₙ`. -/
+scoped syntax:max (name := mkNotation)
+  "⦋" term ("," term)? "⦌" noWs subscriptTerm : term
```

Comment:
```suggestion
  "⦋" term "⦌" noWs subscriptTerm (noWs "'" term)? : term
```
This mirrors the notation of getElem optional proofs.

Reply:
I like the potential for mirroring `getElem` here, but I wasn't able to get this exact syntax working. From [these](https://github.com/leanprover/lean4/blob/4877e8403133097a5c199a8d84ad52d0fcd1c200/tests/lean/run/atomValidation.lean#L49-L63) tests, it seems this is an intentional restriction.

Alternatives could include using a double single quote, `"''"`, using a similar unicode character (see [here](https://github.com/leanprover/lean4/blob/4877e8403133097a5c199a8d84ad52d0fcd1c200/tests/lean/run/atomValidation.lean#L116-L120)), or moving the single quote to the other side of the subscript, `"⦌'"`.

Reply:
I am not sure what the best next step is here. I am happy to change the syntax, but I haven't figured out a way to get the suggested syntax working and none of the alternatives jump out at me as an obvious fit.

Any suggestions, @digama0 @joelriou?

Reply:
I think the best option would be to just avoid using this form altogether and ensure the proof is always implicit. But if we need a syntax for technical reasons, I guess we can just keep it as is. (Other alternatives include more heavyweight syntax like `[n]₂ (h := by decide)`.)